include/circt/Dialect/Seq/SeqPasses.td

```diff
@@ -51,4 +51,10 @@ def ExternalizeClockGate: Pass<"externalize-clock-gate", "mlir::ModuleOp"> {
   ];
 }
 
+def LowerSeqShiftReg : Pass<"lower-seq-shiftreg", "hw::HWModuleOp"> {
```

Comment:
There is a significant cost to running passes.  We don't want to wind up in a situation where we have a lowering pass per operation.

Reply:
This was introduced as a separate pass motivated by the possibility of the shift registers being lowered to other seq operations. However, it's obviously also valid to do that during e.g. seq-to-sv, and just let the lowering patterns apply recursively.